src/users/users.controller.ts

```diff
@@ -5,18 +5,20 @@ import validationMiddleware from "../middleware/validation.middleware";
 import CreateUserDto from "./users.dto";
 import UserNotFoundException from "../exceptions/UserNotFoundException";
 import DbInstance from "../db/db.instance";
+import Controller from "../interfaces/controller.interface";
 
-class UserController {
+class UserController extends Controller {
   public path = "/users";
   public router = Router();
-  private users;
+  private users: DbInstance;
 
   constructor() {
+    super();
     this.initializeRoutes();
     this.users = new DbInstance();
   }
 
-  private initializeRoutes() {
```

Comment:
u can make it private :)

Reply:
abstract function cant be private compiler says